File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int one_students = 0;
        int zero_students = 0;
        int i = 0;
        for(int n: students){
            if(n == 1){
                one_students+=1;
            }else{
                zero_students+=1;
            }
        }
        while(i < sandwiches.length){
            int curr = sandwiches[i];
            if((curr == 0 && zero_students == 0) || (curr == 1 && one_students == 0)){
                return one_students + zero_students;
            }else if(curr == 0){
                i+=1;
                zero_students -= 1;
            }else{
                i+=1;
                one_students -= 1;
            }
        }

        return 0;
    }
}

File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        
        c = 0
        s = 0

        for val in students:
            if val == 0:
                c+=1
            
            else:
                s += 1
        
        for val in sandwiches:
            if val == 0 and c == 0 or val == 1 and s == 0:
                return c + s
            
            elif val == 0:
                c -= 1
            else:
                s -= 1
        return c + s
